tests/integration/integration_test.go

```diff
@@ -644,3 +644,31 @@ func TestExcludeProtected(t *testing.T) {
 	stdout, _ = e.RunCommand("pulumi", "destroy", "--skip-preview", "--yes", "--exclude-protected")
 	assert.Contains(t, stdout, "There were no unprotected resources to destroy. There are still 7")
 }
+
+func TestInvalidPluginError(t *testing.T) {
+	t.Parallel()
+
+	e := ptesting.NewEnvironment(t)
+	defer func() {
+		if !t.Failed() {
+			e.DeleteEnvironment()
+		}
+	}()
+
+	pulumiProject := `
+name: invalid-plugin
+runtime: yaml
+description: A Pulumi program referencing an invalid plugin.
+plugins:
+  providers:
+    - name: fakeplugin
+      bin: ./does/not/exist/bin # key should be 'path'
```

Comment:
Why isn't something like this getting picked up by the schema check?

Reply:
Can you link where the schema check your referring to is?

Reply:
This file https://github.com/pulumi/pulumi/blob/master/sdk/go/common/workspace/project.json

I thought we validated against that when loading project files.
Looks like pluginOptions is missing a "additionalProperties: false" clause.

Reply:
Looks like the validation does run. Those fields just weren't marked required.
If you revert the changes to pkg/ and make the following change to the schema,
it errors instead of panicking.

```diff
diff --git a/sdk/go/common/workspace/project.json b/sdk/go/common/workspace/project.json
index 822122748..ef0d1e23b 100644
--- a/sdk/go/common/workspace/project.json
+++ b/sdk/go/common/workspace/project.json
@@ -238,6 +238,7 @@
         "pluginOptions":{
             "title":"PluginOptions",
             "type":"object",
+            "required":["name","path"],
             "properties":{
                 "name":{
                     "type":"string",
```

New error:

```
% go test -v -run TestInvalidPluginError -tags all
=== RUN   TestInvalidPluginError
=== PAUSE TestInvalidPluginError
=== CONT  TestInvalidPluginError
    environment.go:96: Created new test environment:  /var/folders/12/gsh1s5sj4ds4mgmdyg4b5h400000gn/T/test-env1276381302
    pulumi.go:44: Running command git init
    integration_test.go:670: Running command pulumi stack init invalid-resources
    integration_test.go:670: Run Error: exit status 255
    integration_test.go:670: STDOUT:
    integration_test.go:670: STDERR: error: could not create stack: getting stack tags: error loading project "/private/var/folders/12/gsh1s5sj4ds4mgmdyg4b5h400000gn/T/test-env1276381302/Pulumi.yaml": could not validate '/private/var/folders/12/gsh1s5sj4ds4mgmdyg4b5h400000gn/T/test-env1276381302/Pulumi.yaml': 2 errors occurred:
                * #/plugins/providers/0: doesn't validate with '/$defs/pluginOptions'
                * #/plugins/providers/0: missing properties: 'path'


    integration_test.go:670: Ran command pulumi args [stack init invalid-resources] and expected success. Instead got failure.
--- FAIL: TestInvalidPluginError (0.15s)
FAIL
exit status 1
FAIL    github.com/pulumi/pulumi/tests/integration      0.558s
```

I don't know if that's better, though.

Reply:
Adding "required" is a good call, I'd _also_ add "additionalProperties: false". I think that error is fine, if your project is malformed pulumi should generally just error out that it doesn't understand the project file.